app/javascript/src/utils/compiler/parameterObjects.js

```diff
@@ -0,0 +1,70 @@
+import { completionsMap } from "../../stores/editor"
+import { getClosingBracket, getPhraseFromIndex, replaceBetween, splitArgumentsString } from "../parse"
+import { get } from "svelte/store"
+
+export function evaluateParameterObjects(joinedItems) {
+  let moreAvailableObjects = true
+  let safety = 0
+
+  while (moreAvailableObjects) {
+    const parameterObject = getFirstParameterObject(joinedItems)
+
+    if (!parameterObject || safety > 10_000) {
+      moreAvailableObjects = false
+      continue
+    }
+
+    joinedItems = replaceParameterObject(joinedItems, parameterObject)
+    safety++
+  }
+
+  return joinedItems
+}
+
+export function getFirstParameterObject(content) {
+  const regex = /[a-z]\s*\(\s*{/
+  const match = content.match(regex)
+
+  if (!match) return null
+
+  let end = getClosingBracket(content, "{", "}", match.index - 1)
+  if (end === -1) end = match.index + match.length
+
+  const start = match.index + match[0].indexOf("{")
+  const phrase = getPhraseFromIndex(content, match.index)
+  const completion = get(completionsMap).find(item => item.args_length && item.label.replace(" ", "") === phrase.replace(" ", ""))
+  const string = content.slice(match.index + match[0].length, end).trim()
+  const splitParameters = splitArgumentsString(string)
+  const given = {}
+
+  splitParameters.forEach(item => {
+    const [key, value] = item.split(/:(.*)/s)
+    given[key.trim()] = (value || "").trim()
+  })
+
+  const result = { start, end, given }
+
+  // Return a false object to replace contents of unfound phrase
+  if (!completion) return {
+    ...result,
+    phraseParameters: [],
+    phraseDefaults: []
+  }
```

Comment:
Why would you want to replace the parameters if no completion is found?

Currently this would convert
```
NonExistant({ Foo: bar });
```

into
```
NonExistant();
```

Wouldn't it be less aggresive to just try and find the next parameter object? Even if the syntax is wrong for Workshop, I think the original code should still stay.

Reply:
Yeah the problem was that the while loop finds the first result, so if you don't replace it it will find it again. I will try and find a solution, maybe I can start the loop from a later index or something

Reply:
Use [exec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) instead of match

Reply:
It was a bit trickier than that. The while loop is separate from the regex since the function is also used for one-offs like in the linting.
I opted to in the while loop check if the object was found, and if so set the start index from the index of that match. This index is given to the function and the string is sliced from there so it always starts after that match.
So if the first loop is `Fake({ a: b }) Real({ c: d })` than the second loop starts at `({ a: b }) Real({ c: d })`, and that offset is added to the start and end index of the second result.